mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ConfigRefDeclarationEnricher.java

```diff
@@ -139,6 +143,10 @@ private boolean canBeUsedImplicitly(ParameterizedDeclaration<?> parameterized) {
     return parameterized.getAllParameters().stream().noneMatch(ParameterDeclaration::isRequired);
   }
 
+  private static ExpressionSupport expressionSupport(ComponentDeclaration<?> componentDeclaration) {
+    return componentDeclaration instanceof OperationDeclaration ? SUPPORTED : NOT_SUPPORTED;
```

Comment:
Should this be support for every extension? Would this work for an XML-SDK extension?

Reply:
The idea was for all of this to work regardless of the SDK flavor, yes.
From what I understand, it should work the same, because most changes have been on the `extensions-support` module except for the tests (which were created using a Java SDK extension). If we have concerns about this I guess I could try to add similar tests using an extension written with the XML SDK.
Eventually we will also add tests using operations written with the new Mule SDK, but we need some other things before that.

Reply:
Ok, I suggest you try an example with the XML-SDK. I am approving this one anyways.

Reply:
Good catch! I created the tests and it didn't work due to the macro expansion nature of the properties for the XML SDK.

The `ModuleOperationMessageProcessor` used by the XML SDK does not even extend from `ExtensionComponent`.

As an experiment I created an [additional PR](https://github.com/mulesoft/mule/pull/11560) that could be merged to the one we already have if we are interested in adding support for that. It didn't require too many changes, but there is a little bit of repeated logic from what we have in the `ValueResolverFactory` and also some validations we have at the `ExtensionComponent` level.

Reply:
this should be java sdk specific. Do not add additional behavior to XML SDK